server/grpc_service.go

```diff
@@ -1266,10 +1266,25 @@ func (s *GrpcServer) GetRegion(ctx context.Context, request *pdpb.GetRegionReque
 	if rc == nil {
 		return &pdpb.GetRegionResponse{Header: s.notBootstrappedHeader()}, nil
 	}
-	region := rc.GetRegionByKey(request.GetRegionKey())
+	var region *core.RegionInfo
+	// allow region miss temporarily if this key can't be found in the region tree.
+	for retry := 0; retry <= 10; retry++ {
+		region = rc.GetRegionByKey(request.GetRegionKey())
+		if region != nil {
+			break
+		}
+		select {
+		case <-ctx.Done():
+			break
```

Comment:
You should name the `for` loop, and break this loop. Now, it just break out the `select`.